### PR TITLE
WIP 44 Consolidate and Optimize Imports and Build System

### LINKED ISSUE #44: rpm creation enhancements
The rpm built from this repo should:

1. build in jenkins
2. appear on the Github Releases page as a downloadable artifact
3. not overwrite user config when installed/upgrade

### PR DESCRIPTION
**Pull request by rustydb**
### Summary and Scope

Description: This pull-request satisfies #44 , removing several pieces of the build-system in favor of a simpler model. The simpler model is:
- Build and run a native Python 3.10 application
- Add OS Packaging for that native Python application
- Simplify the `Dockerfile`
- Move `make tests` into a GitHub action to expedite the Jenkins Pipeline builds

The intent here is to also optimize and remove dependencies, making the application smaller. Additionally leveraging PyWheels to include dependencies _with_ the Python application itself. This will remove the need for needing to duplicate dependency lists between PyPi and the OS.

**This drastically changes the root of the repository, it will look very different (less files, and a few new files).** Please carefully review the structure change, and the RPM and Docker Image artifacts.

PR checklist (you may replace this section):
- [ ] I have run `nox` locally and all tests, linting, and code coverage pass.
- [x] I have added new tests to cover the new code
- [x] My code follows the style guidelines of this project
- [ ] If adding a new file, I have updated pyinstaller.py
- [ ] I have updated the appropriate Changelog entries in readme.md
- [x] I have incremented the version in the readme.md
- [x] I have incremented the version in `canu/.version`

### Issues and Related PRs

* Resolves #44 

### Testing

Tested on:

* List of virtual system tested.
* List of physical systems tested.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+include canu/.version
+include README.md
+include canu/canu.yaml
+include canu/validate/switch/config/*.yaml
+include network_modeling/schema/*.json
+include network_modeling/schema/*.yaml
+include network_modeling/models/*.yaml
+recursive-include network_modeling/configs/templates *
+recursive-exclude canu/test
```
Comment (rustydb) [inline MANIFEST.in:9]: This line is bogus, it doesn't work and should be deleted.

File: Makefile [hunk 2]
```diff
@@ -37,10 +37,21 @@ source_path := ${build_dir}/SOURCES/${source_name}.tar.bz2
 all : prepare binary test rpm
 rpm: rpm_package_source rpm_build_source rpm_build
 
+python:
```
Comment (rustydb) [inline Makefile:40]: Could be optimized.

**Comment by rustydb** [comment]
Picking this up off/on while I'm on leave, we can look at merging officially once I'm off leave.

File: Makefile (hunk 2 above)
Comment (trad511) [reply Makefile:40]: yah i wouldn't stress on any of the RPM or pyinstaller stuff.  It's going to be ripped out in full in CSM 1.3.

**Comment by rustydb** [comment]
Still fighting through `nox`, I might need help debugging it.

**Comment by rustydb** [comment]
I'll freeze the modules in setup.py in the morning

**Comment by rustydb** [comment]
I believe I've found the root of my test failures, I'm working on these.

Where `test_file_directory = Path(__file__).resolve().parent` exists I need to refactor to `test_file_directory = Path(__file__).resolve()`.

**Comment by rustydb** [comment]
Closing and redoing.
The redo will just be the python improvement, sans build things.

**Pull request closed without merging**